app/models/user.rb

```diff
@@ -5,6 +5,10 @@ def initialize(client)
 
   def follow_list(user, list)
     users_to_follow = @client.list_members(user, list, skip_status: true, include_entities: false).to_a
+    client_id = @client.user.id
+    users_to_follow.each do |new_user|
+      users_to_follow.delete(new_user) if client_id == new_user.id
+    end
```

Comment:
You appear to have re-implemented [`Array#delete_if`](http://ruby-doc.org/core-2.3.1/Array.html#method-i-delete_if`)/[`Array#reject!`](http://ruby-doc.org/core-2.3.1/Array.html#method-i-reject).